torch_geometric/nn/models/revgnn.py

```diff
@@ -0,0 +1,288 @@
+        with torch.no_grad():
+            # Need edge_index and other args
+            inputs_inverted = ctx.fn_inverse(*(outputs + inputs[1:]))
+            # Clear memory from outputs
+            for element in outputs:
+                element.storage().resize_(0)
+
+            if not isinstance(inputs_inverted, tuple):
+                inputs_inverted = (inputs_inverted, )
+            for element_original, element_inverted in zip(
+                    inputs, inputs_inverted):
+                element_original.storage().resize_(
+                    int(np.prod(element_original.size())))
+                element_original.set_(element_inverted)
+
+        # Compute gradients with grad enabled
+        with torch.set_grad_enabled(True):
+            detached_inputs = []
+            for element in inputs:
+                if isinstance(element, torch.Tensor):
+                    detached_inputs.append(element.detach())
+                else:
+                    detached_inputs.append(element)
+            detached_inputs = tuple(detached_inputs)
+            for det_input, requires_grad in zip(detached_inputs,
+                                                ctx.input_requires_grad):
+                if isinstance(det_input, torch.Tensor):
+                    det_input.requires_grad = requires_grad
+            temp_output = ctx.fn(*detached_inputs)
+
+        if not isinstance(temp_output, tuple):
+            temp_output = (temp_output, )
+        filtered_detached_inputs = tuple(
+            filter(
+                lambda x: x.requires_grad
+                if isinstance(x, torch.Tensor) else False,
+                detached_inputs,
+            ))
+        gradients = torch.autograd.grad(
+            outputs=temp_output,
+            inputs=filtered_detached_inputs + ctx.weights,
+            grad_outputs=grad_outputs,
+        )
+
+        input_gradients = []
+        i = 0
+        for rg in ctx.input_requires_grad:
+            if rg:
+                input_gradients.append(gradients[i])
+                i += 1
+            else:
+                input_gradients.append(None)
+
+        gradients = tuple(input_gradients) + gradients[-len(ctx.weights):]
+
+        return (None, None, None, None) + gradients
+
+
+class InvertibleModule(nn.Module, ABC):
+    r"""An abstract class for implementing Invertible Modules.
+
+    Args:
+        disable (bool, optional): This disables using the InvertibleFunction.
+            Therefore, it executes functions without memory savings.
+        num_bwd_passes (int, optional):
+            Number of backward passes to retain a link with the output.
+            After the last backward pass the output is discarded
+            and memory is freed.
+    """
+    def __init__(self, disable=False, num_bwd_passes=1):
+        super().__init__()
+        self.disable = disable
+        self.num_bwd_passes = num_bwd_passes
+
+    def forward(self, x: Tensor, edge_index: Adj, *args):
```

Comment:
```suggestion
    def forward(self, *args, **kwargs):
```
Not sure why we need to expose `x` and `edge_index` explicitly.